addons/civilian/functions/fnc_initCity.sqf

```diff
@@ -0,0 +1,35 @@
+#include "script_component.hpp"
+/*
+ * Author: 3Mydlo3
+ * Function initializes given city with module, civilians and vehicles.
+ *
+ * Arguments:
+ * 0: City to initialize <CONFIG/LOCATION>
+ *
+ * Return Value:
+ * 0: City logic <LOGIC>
+ *
+ * Example:
+ * [[player] call afsk_common_fnc_nearestLocation] call afsk_common_fnc_initCity
+ *
+ * Public: No
+ */
+
+params ["_city"];
+
+if (_city isEqualType configNull) then {
+    _city = [getArray (_x >> 'position'), 10] call EFUNC(common,nearestLocation);
+};
+
+// Create city logic
+private _cityLogic = (createGroup sideLogic) createUnit ["LOGIC", position _city, [], 0, "CAN_COLLIDE"];
```

Comment:
No, I cannot create own variables on map locations.
> They can use setVariable and getVariable, but this only works with user made locations
> Source: https://community.bistudio.com/wiki/Location

I wanted to be able to find city logic by position (as it can have it's own variables unlike standard locations). Can it be done efficiently with CBA namespaces?

Reply:
You can search for `CBA_NamespaceDummy` objects, but TBH I don't like this approach with searching for objects near location.

You could create another namespace to hold references to location related namespaces, so instead of searching for them like:
```sqf
private _logic = nearestObject [position _nearestTown, "LOGIC"];
```
you would do:
```sqf
private _namespace = GVAR(cities) getVariable [className _nearestTown, objNull]; (or locationNull if local namespace)
```